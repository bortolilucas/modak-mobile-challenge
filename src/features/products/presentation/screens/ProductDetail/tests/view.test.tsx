import * as states from '@/features/products/presentation/screens/ProductDetail/tests/componentStates';
import { testSnapshots } from '@/testing/snapshots';

describe('ProductDetail view', () => {
  testSnapshots(states);
});
