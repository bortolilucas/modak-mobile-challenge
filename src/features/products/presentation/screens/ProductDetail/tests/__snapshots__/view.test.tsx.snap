// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail view should match Default snapshot 1`] = `
<RCTScrollView>
  <View>
    <RNCSafeAreaView
      edges={
        {
          "bottom": "additive",
          "left": "off",
          "right": "off",
          "top": "off",
        }
      }
      style={
        {
          "gap": 16.675,
          "paddingHorizontal": 41.66666666666667,
          "paddingTop": 4.16875,
        }
      }
    >
      <View
        style={
          {
            "borderColor": "#bec9c6",
            "borderRadius": 20.833333333333336,
            "borderWidth": 2.0833333333333335,
            "height": 333.5,
            "marginBottom": 4.16875,
            "width": "100%",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            {
              "uri": "",
            }
          }
          style={
            {
              "height": "100%",
              "width": "100%",
            }
          }
        />
        <View
          style={
            [
              {
                "backgroundColor": "#006a61",
                "borderTopLeftRadius": 12.5,
                "justifyContent": "center",
                "paddingHorizontal": 16.666666666666668,
                "paddingVertical": 3.1265625000000004,
              },
              {
                "borderBottomRightRadius": 12.5,
                "bottom": 0,
                "position": "absolute",
                "right": -2.0833333333333335,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 25,
                  "fontWeight": "500",
                  "lineHeight": 33.333333333333336,
                },
                {
                  "color": "#fff",
                },
                {
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            category-1
          </Text>
        </View>
      </View>
      <Text
        style={
          [
            {
              "fontSize": 50,
              "fontWeight": "700",
              "lineHeight": 62.50000000000001,
            },
            {
              "color": "#191c1b",
            },
            undefined,
          ]
        }
      >
        Mock product
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 20.833333333333336,
            "marginBottom": 8.3375,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 8.333333333333334,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={33.333333333333336}
            bbWidth={33.333333333333336}
            focusable={false}
            height={33.333333333333336}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={33.333333333333336}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGPath
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                fill={
                  {
                    "payload": 4282337607,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <Text
            style={
              [
                {
                  "fontSize": 29.166666666666668,
                  "fontWeight": "500",
                  "lineHeight": 41.66666666666667,
                },
                {
                  "color": "#3f4947",
                },
                undefined,
              ]
            }
          >
            1
          </Text>
        </View>
        <Text
          style={
            [
              {
                "fontSize": 29.166666666666668,
                "fontWeight": "500",
                "lineHeight": 41.66666666666667,
              },
              {
                "color": "#3f4947",
              },
              undefined,
            ]
          }
        >
          Brand: 
        </Text>
      </View>
      <Text
        style={
          [
            {
              "fontSize": 29.166666666666668,
              "fontWeight": "500",
              "lineHeight": 41.66666666666667,
            },
            {
              "color": "#191c1b",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "gap": 25,
            "marginTop": 20.84375,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#dfedea",
              "borderRadius": 20.833333333333336,
              "flexDirection": "row",
              "gap": 20.833333333333336,
              "paddingHorizontal": 25,
              "paddingVertical": 20.84375,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={33.333333333333336}
            bbWidth={33.333333333333336}
            fill="none"
            focusable={false}
            height={33.333333333333336}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={33.333333333333336}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M11 21.73a2 2 0 0 0 2 0l7-4A2 2 0 0 0 21 16V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73z"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4278217313,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              />
              <RNSVGPath
                d="M12 22V12"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4278217313,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              />
              <RNSVGPath
                d="M3.29 7 12 12 20.71 7"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4278217313,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              />
              <RNSVGPath
                d="m7.5 4.27 9 5.15"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "payload": 4278217313,
                    "type": 0,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <View>
            <Text
              style={
                [
                  {
                    "fontSize": 25,
                    "fontWeight": "500",
                    "lineHeight": 33.333333333333336,
                  },
                  {
                    "color": "#3f4947",
                  },
                  undefined,
                ]
              }
            >
              Stock
            </Text>
            <Text
              style={
                [
                  {
                    "fontSize": 29.166666666666668,
                    "fontWeight": "500",
                    "lineHeight": 41.66666666666667,
                  },
                  {
                    "color": "#191c1b",
                  },
                  {
                    "textTransform": "capitalize",
                  },
                ]
              }
            >
              1 Unit
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#dfedea",
              "borderRadius": 20.833333333333336,
              "flexDirection": "row",
              "gap": 20.833333333333336,
              "paddingHorizontal": 25,
              "paddingVertical": 20.84375,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={33.333333333333336}
            bbWidth={33.333333333333336}
            fill="none"
            focusable={false}
            height={33.333333333333336}
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="#006a61"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 33,
                  "width": 33,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={33.333333333333336}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4278217313,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="2"
            >
              <RNSVGPath
                d="M12 15v-3.014"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M16 15v-3.014"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M20 6H4"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M20 8V4"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M4 8V4"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M8 15v-3.014"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGRect
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                height="7"
                rx="1"
                width="18"
                x="3"
                y="12"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <View>
            <Text
              style={
                [
                  {
                    "fontSize": 25,
                    "fontWeight": "500",
                    "lineHeight": 33.333333333333336,
                  },
                  {
                    "color": "#3f4947",
                  },
                  undefined,
                ]
              }
            >
              Dimensions (w x d x h)
            </Text>
            <Text
              style={
                [
                  {
                    "fontSize": 29.166666666666668,
                    "fontWeight": "500",
                    "lineHeight": 41.66666666666667,
                  },
                  {
                    "color": "#191c1b",
                  },
                  {
                    "textTransform": "capitalize",
                  },
                ]
              }
            >
              0 × 0 × 0
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#006a61",
            "borderRadius": 52.083333333333336,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 52.109375,
            "minHeight": 75.03750000000001,
            "opacity": 1,
            "padding": 25,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 29.166666666666668,
                "fontWeight": "500",
                "lineHeight": 41.66666666666667,
              },
              {
                "color": "#fff",
              },
              undefined,
            ]
          }
        >
          Add purchase reminder
        </Text>
      </View>
    </RNCSafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`ProductDetail view should match Loading snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="#006a61"
    size="large"
  />
</View>
`;
